convergedtarkus/semver_parser
# Add Greater and less than Support

## Problem
There is no support for the <, >, <= and >= comparison operators. 

## Solution
Added support for the <, >, <= and >= comparison operators. These operators are also supported in the example app now.

## Implementation Notes
These new operators are implemented in a similar fashion to the ==/!= operators. Semantic versions are compared less to right for the first difference (major>minor>patch) as defined in the Semantic version spec.
The code for ==/!= has been modified to share code with the new <, >, <= and >= operators. This allows easier support for the <= and >= operators without duplicating code.

## Tests Added
Added tests over precedence. Also expanded existing these to also check these new operators (E.X. The equality tests verify < and > are false and <= and >= are true).
Also added some new tests over the 0.0.0 version and a few more corner cases.

## Testing Instructions
- Test new operators (<, >, <= and >=)
- Test == and != operators (code was refactored)
- Verify example app can parse and use all 6 operators
- Verify example usage information includes <, >, <= and >=
